### DISCUSSION TITLE
Resimulating old replays

### DISCUSSION THREAD

**Discussion by lemoncove** [ppy/osu#23935] [Q&A] [answered]
#23892 migrated old scores to the new scoring algorithm. Unfortunately, since the new algorithm (#23638) is non-recalculable, new total score values are best-effort estimations going off only end-screen statistics. The estimations are pretty good but skew a little higher than the ground-truth total scores, which can only be calculated by actually running through the replay. Obviously going through every replay would be extremely computationally expensive and would be unsuitable for a simple migration, but I do wish it were an option for those who really want to get their local leaderboards as accurate possible—something to run overnight, say.

I tried writing a script that would achieve this but found it difficult to wrap my head around the use of `DrawableRuleset`, or even puppeteering a `ReplayPlayer`. Any guidance would be much appreciated.

**Comment by peppy** [comment, marked as answer]
This is not going to happen sorry (out of scope, too much effort for a very niche user base that cares to this level). If you need this, you can manage it yourself.

The simplest way to implement it would be to have it run whenever you watch a replay (at the results screen or end of player) then watch the replays that you want converted one by one, IMO.

**Reply by lemoncove** [reply]
Thank you, I was mostly hoping for guidance on how I could approach doing something like this. Putting the Realm write in `ReplayPlayer.ImportScore` works perfectly and `OsuGame.PresentScore` (plus `Player.Seek`) can be used to automatically queue replays one after another, if that's useful for anyone else who wants to do the same.
